src/xtgeo/xyz/polygons.py

```diff
@@ -531,6 +572,13 @@ def copy(self):
         mycopy._tname = self._tname
         mycopy._dtname = self._dtname
 
+        if self._attrs:
+            attrs = {}
+            for key, value in self._attrs.items():
+                attrs[key] = value
+
+            mycopy._attrs = attrs
```

Comment:
Same as above I prefer the dict comprehension here
```suggestion
            mycopy._attrs = {key: value for key, value in self._attrs.items()}       
```